src/main/java/carpet/script/utils/ShapeDispatcher.java

```diff
@@ -546,6 +563,129 @@ public long calcKey()
         }
     }
 
+    public static class DisplayedSprite extends ExpiringShape {
+        private final Set<String> required = Set.of("pos");
+        private final Map<String, Value> optional = Map.ofEntries(
+                entry("facing", new StringValue("player")),
+                entry("tilt", new NumericValue(0)),
+                entry("lean", new NumericValue(0)),
+                entry("turn", new NumericValue(0)),
+                entry("height", new NumericValue(1)),
+                entry("width", new NumericValue(1)),
+                entry("obj_size", new NumericValue(1)),
+                entry("light_fromblock", new NumericValue(-999)),
+                entry("light_fromsky", new NumericValue(-999)),
+                entry("needf3b", BooleanValue.FALSE));
+        private boolean isitem;
+
+        @Override
+        protected Set<String> requiredParams() {
+            return Sets.union(Sets.union(super.requiredParams(), required), Set.of(isitem ? "item" : "block"));
+        }
+
+        @Override
+        protected Set<String> optionalParams() {
+            return Sets.union(Sets.union(super.optionalParams(), optional.keySet()),isitem ? Set.of("item_transform_type") : Set.of());
+        }
+
+        public DisplayedSprite(boolean i) {
+            isitem = i;
+        }
+
+        Vec3 pos;
+
+        Direction facing;
+
+        float tilt;
+        float lean;
+        float turn;
+        float size;
+        int light_fromblock;
+        int light_fromsky;
+
+        float height;
+        float width;
+        boolean needf3b;
+        CompoundTag blockEntity;
+        BlockState blockState;
+        ItemStack item = null;
+        TransformType item_transform_type;
+
+        @Override
+        protected void init(Map<String, Value> options) {
+            super.init(options);
+            pos = vecFromValue(options.get("pos"));
+            blockState = ((BlockValue) options.getOrDefault("block", BlockValue.AIR)).getBlockState();
```

Comment:
getting the block should happen once and reused.

Reply:
and as suggested make it one argument and make it mandatory, so not expecting AIR as default

Reply:
ok. i fix it.